examples/cms/src/data.ts

```diff
@@ -0,0 +1,166 @@
+import { CollectionItemData, ManagedCollection, ManagedCollectionField, framer } from "framer-plugin"
+import { generateHash, isNotNull } from "./utils"
+import { PLUGIN_KEYS } from "./constants"
+
+export type ManagedCollectionFieldType = ManagedCollectionField["type"]
+
+type DataSourceField = {
+    type: ManagedCollectionFieldType
+}
+
+export interface DataSource {
+    id: string
+    fields: Record<string, DataSourceField>
+    items: Record<string, unknown>[]
+}
+
+export interface FieldConfig {
+    name: string
+    isNew: boolean
+    field: ManagedCollectionField | null
```

Comment:
A field already has a `name` attribute right? Why do we need a separate name in the config? Also, can't we compute `isNew` by comparing the fields on the node with the mutable fields in React state?

Reply:
This is the original name of the field, as set in the API

Reply:
I think it would be more clear if you have the current field and mutable fields as separate state. And then you can compute this instead of having multiple sources of truth